### PR TITLE
feature/npmInstall

### PR DESCRIPTION
**Pull request by GuentherJulian**

### CODE REVIEW

**Review by denise-khuu — DISMISSED** [review]

File: documentation/Functions.md [hunk 1]
```diff
@@ -135,4 +137,13 @@ runServerJava("devonfw/workspaces/main/jump-the-queue/java/jtqj/server", { "star
 startupTime = Time in seconds to wait before checking if the server is running
 port: Port on which the server is running
 path: The URL path on which is checked if the server is running
+
+***
+
+### npmInstall
+#### parameter 
+1. Path to the project where the dependencies from the angular.json file are to be installed.
```
Comment (SchettlerKoehler) [inline documentation/Functions.md:145]: package.json

File: runners/console/index.ts [hunk 2]
```diff
@@ -180,6 +180,16 @@ export class Console extends Runner {
         return result;
     }
 
+    runNpmInstall(step: Step, command: Command): RunResult {
+        let result = new RunResult();
+        result.returnCode = 0;
+
+        let projectPath = path.join(this.getWorkingDirectory(), "devonfw", "workspaces", "main", command.parameters[0]);
+        this.executeDevonCommandSync("npm install", projectPath, result);
```
Comment (SchettlerKoehler) [inline runners/console/index.ts:188]: Is that correct? I would have used executeCommandSync

File: runners/console/index.ts [hunk 3]
```diff
@@ -191,6 +201,7 @@ export class Console extends Runner {
 
         for(let i = 0; i < installedTools.length; i++) {
             if(installedTools[i] == "mvn") installedTools[i] = "maven";
+            if(installedTools[i] == "npm") installedTools[i] = "node";
```
Comment (SchettlerKoehler) [inline runners/console/index.ts:204]: Maybe you should add a map for that

File: runners/katacoda/templates/npmInstall.md [hunk 4]
```diff
@@ -0,0 +1,11 @@
+<%= text; %>
+
+<%= cdCommand; %>
+
+Install the dependencies from the angular.json file.
```
Comment (SchettlerKoehler) [inline runners/katacoda/templates/npmInstall.md:5]: package.json

File: runners/katacoda/templates/npmInstall.md [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+
+`devon npm install`{{execute}}
```
Comment (SchettlerKoehler) [inline runners/katacoda/templates/npmInstall.md:7]: I would have used npm install, but I am not sure if that is correct

File: runners/console/index.ts [hunk 6]
```diff
@@ -331,12 +355,12 @@ export class Console extends Runner {
...
 
-    private executeDevonCommandAsync(devonCommand: string, directory: string, result: RunResult, input?: string): child_process.ChildProcess {
+    private executeDevonCommandAsync(devonCommand: string, directory: string, result: RunResult): child_process.ChildProcess {
         if(this.platform == ConsolePlatform.WINDOWS) {
             let scriptsDir = path.join(this.getWorkingDirectory(), "devonfw", "scripts");
-            return this.executeCommandAsync(scriptsDir + "\\devon " + devonCommand, directory, result, input);
+            return this.executeCommandAsync(scriptsDir + "\\devon " + devonCommand, directory, result);
         } else {
-            return this.executeCommandAsync("~/.devon/devon " + devonCommand, directory, result, input);
+            return this.executeCommandAsync("~/.devon/devon " + devonCommand, directory, result);
```
Comment (SchettlerKoehler) [inline runners/console/index.ts:363]: Denk bitte dran, was wir über die Abhängigkeiten zwischen den Tests besprochen hatten und entferne bitte im init den Ordner ~/.devon

**Review by SchettlerKoehler — APPROVED** [review]

**Pull request merged**